### PR TITLE
docs: Add simplied core internals implementation

### PR DESCRIPTION
**Pull request by ankitchouhan1020**
Created a simplified core implementation doc from Daishi's tweets. Added comments and some description for easy understanding.

### CODE REVIEW

File: docs/guides/core-internals.mdx [hunk 1]
```diff
@@ -0,0 +1,215 @@
+---
+title: Core Internals
+description: A simplified version of the core implementation
+nav: 3.10
+---
+
+This guide is beneficial for those who want to understand the core implementation of the jotai. It is not meant to be a complete guide of the 
+core implementation but rather a simplified version. It is inspired by the collection of tweets by Daishi Kato([@dai_shi](https://twitter.com/dai_shi)).
```
Comment (dai-shi) [inline docs/guides/core-internals.mdx:8]: Maybe we put links to the collection of tweets for reference?

**Review by dai-shi — COMMENTED** [review]
This is great! Thanks for working on this.

File: docs/guides/core-internals.mdx (hunk 1 above)
Comment (ankitchouhan1020) [reply docs/guides/core-internals.mdx:8]: Hi @dai-shi, I have added ref tweet below the implementation. Let me know if this works.

Do we need to add or update any comments in the second implementation ?

Comment (dai-shi) [reply docs/guides/core-internals.mdx:8]: Thanks. It works.

It looks good. If you/we have ideas to improve it, feel free to raise suggestions.

**Review by dai-shi — APPROVED** [review]
Thanks for your contribution! Merging.

File: docs/guides/core-internals.mdx (hunk 1 above)
Comment (ankitchouhan1020) [reply docs/guides/core-internals.mdx:8]: Great! May be we can add a high level architecture overview of its working. Something like redux [does](https://redux.js.org/assets/images/ReduxDataFlowDiagram-49fa8c3968371d9ef6f2a1486bd40a26.gif).

Comment (dai-shi) [reply docs/guides/core-internals.mdx:8]: Something like this? https://twitter.com/dai_shi/status/1380885195549597696

Comment (ankitchouhan1020) [reply docs/guides/core-internals.mdx:8]: Yes, with animating atom's read write operations and how it re-renders all the subscribed components.

Comment (dai-shi) [reply docs/guides/core-internals.mdx:8]: yeah, that would be nice.

**Pull request merged**